core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -4504,14 +4514,23 @@ private RecordCursorFactory generateSelectChoose(QueryModel model, SqlExecutionC
             }
 
             if (index == timestampIndex) {
-                selectMetadata.setTimestampIndex(i);
-                timestampSet = true;
+                // Always prefer the column matching the first ORDER BY column as the designated
+                // timestamp in case of multiple timestamp aliases, e.g. `select ts, ts as ts1, ...`.
+                // That's to choose the optimal plan in generateOrderBy().
+                // Otherwise, prefer columns with aliases matching the base column name, e.g.
+                // prefer `t1.ts as ts` over `t1.ts as ts2`.
+                if (Chars.equalsIgnoreCaseNc(queryColumn.getAlias(), firstOrderByColumn)
+                        || Chars.equalsIgnoreCase(queryColumn.getAlias(), timestampName, timestampNameDot + 1, timestampName.length())
+                        || !timestampSet) {
+                    queryMetadata.setTimestampIndex(i);
+                    timestampSet = true;
+                }
```

Comment:
What timestamp will be set to if the execution does not go into the if block? What if `firstOrderByColumn == null`, why do we alter the old behaviour in that case?

Reply:
No, not very clear, better to be separate if tbh.

What if this condition `Chars.equalsIgnoreCase(queryColumn.getAlias(), timestampName, timestampNameDot + 1, timestampName.length())`  is also false, but the old condition `index == timestampIndex` is true, why do we change the behaviour in this situation? Or is it impossible?